pkg/firewall/firewall.go

```diff
@@ -51,9 +51,9 @@ func NewStakeOrActiveKeepPolicy(
 		minimumStakePolicy:          coreFirewall.MinimumStakePolicy(stakeMonitor),
 		authorizedOperatorsCache:    cache.NewTimeCache(authorizationCachePeriod),
 		nonAuthorizedOperatorsCache: cache.NewTimeCache(authorizationCachePeriod),
-		activeKeepMembersCache:      cache.NewTimeCache(activeKeepCachePeriod),
-		noActiveKeepMembersCache:    cache.NewTimeCache(activeKeepCachePeriod),
-		keepInfoCache:               newKeepInfoCache(),
+		activeKeepMembersCache:      cache.NewTimeCache(activeKeepMemberCachePeriod),
+		noActiveKeepMembersCache:    cache.NewTimeCache(activeKeepMemberCachePeriod),
+		keepInfoCache:               newKeepInfoCache(activeKeepMemberCachePeriod),
```

Comment:
I just had a small epiphany: I don't think this caching should be happening in the firewall package. I think it should be happening in the chain implementation.

Reply:
I had the same thought at some point but I landed with caching here, given that not all the information we cache is invariant and it may depend on the particular implementation whether they are fine with some information being out of date or not.

For example, we can cache in the chain implementation information that a keep is inactive. We can also cache keep members - this information never changes. On the other hand, we should not be caching information whether keep is active as this may change at any moment. Firewall will be fine with having this information out of date for some time, but tBTC chain extensions, for instance, may want to have it always up-to-date.

Reply:
Hmmm… I see some of this yeah.  I guess a better way to put it might be, “we might be conflating things that we want the firewall to cache and things that we want the chain to cache”. For example, keep active status can probably be cached at the chain level since it's immutable once settled.

But… Not something to dig into too much right now I think.